src/components/textfieldButtons/js/TextfieldButtons.js

```diff
@@ -0,0 +1,153 @@
+                    incrementCoefficient: -1
+                }
+            }
+        },
+        selectors: {
+            textfield: ".flc-textfieldButtons-field",
+            increaseButton: ".fl-increase-button",
+            decreaseButton: ".fl-decrease-button"
+        },
+        model: {
+            value: null
+        },
+        modelRelay: {
+            target: "value",
+            singleTransform: {
+                type: "fluid.transforms.limitRange",
+                input: "{that}.model.value",
+                min: "{that}.options.range.min",
+                max: "{that}.options.range.max"
+            }
+        },
+        range: {
+            min: 0,
+            max: 100
+        }
+    });
+
+    fluid.defaults("fluid.textfieldButtons.textfield", {
+        gradeNames: ["fluid.viewComponent"],
+        range: {}, // should be used to specify the min, max range e.g. {min: 0, max: 100}
+        modelRelay: {
+            target: "value",
+            singleTransform: {
+                type: "fluid.transforms.stringToNumber",
+                input: "{that}.model.stringValue"
+            }
+        },
+        modelListeners: {
+            value: {
+                "this": "{that}.container",
+                "method": "val",
+                args: ["{change}.value"]
+            },
+            // TODO: This listener is to deal with the issue that, when the input field receives a invalid input such as a string value,
+            // ignore it and populate the field with the previous value.
+            // This is an area in which UX has spilled over into our model configuration, which to some extent we should try to prevent.
+            // Whenever we receive a "change" event or some other similar checkpoint, if these updates occurred any faster, the user would
+            // be infuriated by being unable to type into the field. This situation doesn't occur at the moment because the change event is
+            // only fired when users leave the input feild. At the very least, we need to give a namespace to this listener - unfortunately
+            // the current dataBinding implementation will ignore it. Having this listener here represents an interaction decision rather
+            // than an implementation decision. This issue needs to be revisited.
+            stringValue: {
+                "this": "{that}.container",
+                "method": "val",
+                args: ["{that}.model.value"]
+            }
+        },
+        listeners: {
+            "onCreate.bindChangeEvt": {
+                "this": "{that}.container",
+                "method": "change",
+                "args": ["{that}.setModel"]
+            }
+        },
+        invokers: {
+            setModel: {
+                changePath: "stringValue",
+                value: "{arguments}.0.target.value"
+            }
+        }
+    });
+
+    fluid.defaults("fluid.button", {
```

Comment:
The implementation seems to be for a specific type of button, one that increments/decrements a value. `fluid.button` seems like it should be for a generic button. This should probably be called something like `fluid.stepper.button`, `fluid.button.stepper` or something like that.